servant-client/test/Servant/Client/TestServer.hs

```diff
@@ -0,0 +1,12 @@
+{-# LANGUAGE CPP                    #-}
+
+module Servant.Client.TestServer (
+  buildTestServer,
+  withTestServer,
+)where
+
```

Comment:
Here seems like a good way of explaining how the tests are run in each situation. Then the actual modules can just point here.